### PR TITLE
rhel: allow deactivation of ContainerAPI fetching

### PR DESCRIPTION
**Pull request by BradLugo**
## Description

There's logic in the RHEL repository scanner to disable reaching out to the internet, but we currently set a default API endpoint during Configure to prevent that condition. These changes introduce a DisableAPI flag to allow the deactivation of the fetching from the Red Hat Container API.

### CODE REVIEW

File: rhel/repositoryscanner_test.go [hunk 1]
```diff
@@ -100,6 +122,50 @@ func TestRepositoryScanner(t *testing.T) {
...
+					Key:  repositoryKey,
+					CPE:  cpe.MustUnbind("cpe:/o:redhat:enterprise_linux:7::server"),
+				},
+				{
+					Name: "cpe:/o:redhat:enterprise_linux:8::server",
+					Key:  repositoryKey,
+					CPE:  cpe.MustUnbind("cpe:/o:redhat:enterprise_linux:8::server"),
+				},
+			},
+			cfg:       &RepositoryScannerConfig{DisableAPI: true, API: esrv.URL, Repo2CPEMappingURL: "/", Repo2CPEMappingFile: f.Name()},
```
Comment (RTann) [inline rhel/repositoryscanner_test.go:166]: Can you add a test where we can't get any repo data because there is no file and `DisableAPI: true`?

Comment (BradLugo) [reply rhel/repositoryscanner_test.go:166]: I'm not entirely following, but I tried to add what I thought you were getting at. If the new changes don't seem quite right, can you rephrase your comment?

Comment (RTann) [reply rhel/repositoryscanner_test.go:166]: yeah this is what I meant, thanks

File: rhel/repositoryscanner_test.go [hunk 2]
```diff
@@ -31,6 +32,10 @@ func TestRepositoryScanner(t *testing.T) {
 		"rh-pkg-1-1": strings.NewReader(`{"data":[{"cpe_ids":["cpe:/o:redhat:enterprise_linux:8::computenode","cpe:/o:redhat:enterprise_linux:8::baseos"],"parsed_data":{"architecture":"x86_64","labels":[{"name":"architecture","value":"x86_64"}]}}]}`),
 	}
 	mappingData := strings.NewReader(`{"data":{"content-set-1":{"cpes":["cpe:/o:redhat:enterprise_linux:6::server","cpe:/o:redhat:enterprise_linux:7::server"]},"content-set-2":{"cpes":["cpe:/o:redhat:enterprise_linux:7::server","cpe:/o:redhat:enterprise_linux:8::server"]}}}`)
+	mappingDataBytes := new(bytes.Buffer)
```
Comment (RTann) [inline rhel/repositoryscanner_test.go:35]: up to you, but you could also just do `var mappingDataBytes bytes.Buffer`

**Review by RTann — DISMISSED** [review]

File: rhel/repositoryscanner_test.go [hunk 3]
```diff
@@ -55,6 +60,23 @@ func TestRepositoryScanner(t *testing.T) {
 	srv := httptest.NewServer(mux)
 	defer srv.Close()
 
+	esrv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		t.Fatal("external http request invoked when none was expected")
+	}))
+
+	f, err := os.CreateTemp("", "repository-2-cpe.json")
```
Comment (crozzy) [inline rhel/repositoryscanner_test.go:67]: nit but we have `pkg/tmp.NewFile()` and `.Close()` that will close the file and remove it.

File: rhel/repositoryscanner_test.go [hunk 4]
```diff
@@ -55,6 +60,23 @@ func TestRepositoryScanner(t *testing.T) {
...
+	if err != nil {
+		t.Fatal("trying to create repository-2-cpe.json for FromMappingFile test", err)
+	}
+	defer os.Remove(f.Name())
+
+	if _, err := f.Write(mappingDataBytes.Bytes()); err != nil {
+		t.Fatal(fmt.Sprintf("trying to write %s for FromMappingFile test", f.Name()), err)
```
Comment (crozzy) [inline rhel/repositoryscanner_test.go:74]: `t.Fatalf` is useful here (and in the next block)

**Review by crozzy — CHANGES_REQUESTED** [review]
LGTM, just a couple of nits

File: rhel/repositoryscanner_test.go (hunk 3 above)
Comment (RTann) [reply rhel/repositoryscanner_test.go:67]: another option is using `t.TempDir()` which will auto-close once the test is done

Comment (BradLugo) [reply rhel/repositoryscanner_test.go]: A combination of the two seems pretty nice

File: rhel/repositoryscanner_test.go (hunk 4 above)
Comment (BradLugo) [reply rhel/repositoryscanner_test.go]: 🤦 I swear I've written Go before

**Comment by BradLugo** [comment]
Thanks for the feedback, yall!
